include/envoy/stats/stats_matcher.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Stats {
+
+class StatsMatcher {
+public:
+  virtual ~StatsMatcher() {}
+
+  /**
+   * Take a metric name and report whether or not it should be instantiated.
+   * @param name std::string& a name of Stats::Metric (Counter, Gauge, Histogram).
```

Comment:
I don't think we usually put type types into the `@param` declarations. Whereas from what I've seen we typically do put them in `@return` declarations. But as this is not enforced by any toolchain I've seen run, I don't think it's very consistent at this point. I'll leave it at that :)

Reply:
I'll try to clean this up across StatsMatcher/Impl, but I also don't know what best practice is here.2